### PR TITLE
pro plan options

### PR DESCRIPTION
**Pull request by gerouvi**
Added the options of the Pro Plan with the modal to send an email requiring them or scheduling an appointment

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,2 +1,2 @@
 VOCDONI_ENVIRONMENT=prod
-FEATURES='{"faucet":true,"vote":{"anonymous":true,"overwrite":true,"secret":true},"login":["web3","web2"],"census":["spreadsheet","token","web3"],"languages":["en","es","ca"]}'
+FEATURES='{"faucet":true,"vote":{"anonymous":true,"overwrite":true,"secret":true,"customization":true},"types":["single":true,"multi":true,"approval":true,"participatory":true,"borda":true],"login":["web3","web2"],"census":["spreadsheet","token","web3"],"unimplemented_census":["phone","email","crm","database","digital_certificate","others"],"languages":["en","es","ca"]}'
```
Comment (elboletaire) [inline .env.example:2]: You used an array to define an object in types.

**Review by elboletaire — CHANGES_REQUESTED** [review]
I'll continue reviewing tomorrow...

**Review by jpaulet — APPROVED** [review]

File: vite.config.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ const viteconfig = ({ mode }) => {
   // load env variables from .env files
   process.env = { ...process.env, ...loadEnv(mode, process.cwd(), '') }
 
-  let vocdoniEnvironment = process.env.VOCDONI_ENVIRONMENT
+  let vocdoniEnvironment = process.env.VOCDONI_EºNVIRONMENT
```
Comment (elboletaire) [inline vite.config.ts:14]: What the actual...

File: src/importmeta.d.ts [hunk 3]
```diff
@@ -7,22 +7,43 @@ interface ImportMeta {
...
+        digital_certificate: boolean
+        others: boolean
+      }
+      _voting: {
+        single: boolean
+        multi: boolean
+        approval: boolean
+        participatory: boolean
+        borda: boolean
+      }
```
Comment (elboletaire) [inline src/importmeta.d.ts:46]: I don't see these in features.ts, and they should be defined there, like it's done with `_census`.

**Review by elboletaire — CHANGES_REQUESTED** [review]
A part from the code changes in the review, there are some other visual things that should be addressed:

imatge
These should not be overlapping with the census manager.

Probably related to the wrong features initialization, disabling features does not seem to be working:
imatge
This is the rendered component after setting `FEATURES='{"voting":["single"]}'`

**Comment by jpaulet** [comment]
This branch needs a full review: the "onvote" theme is not working properly, and in the default one, there are some issues. Let's talk about it next week.

**Comment by jpaulet** [comment]
Change the link to book a meeting to: [](https://calendly.com/vocdoni-app/30min)

**Comment by elboletaire** [comment]
@gerouvi remember to update this branch with develop one, otherwise this cannot be merged.

**Comment by jpaulet** [comment]
When trying to send a mail, this error appears: 
image

File: .env.example [hunk 4]
```diff
@@ -1,2 +1,2 @@
 VOCDONI_ENVIRONMENT=prod
-FEATURES='{"faucet":true,"vote":{"anonymous":true,"overwrite":true,"secret":true},"login":["web3","web2"],"census":["spreadsheet","token","web3"],"languages":["en","es","ca"]}'
+FEATURES='{"faucet":true,"vote":{"anonymous":true,"overwrite":true,"secret":true,"customization":true},"types":{"single":true,"multi":true,"approval":true,"participatory":true,"borda":true},"login":["web3","web2"],"census":["spreadsheet","token","web3"],"unimplemented_census":["phone","email","crm","database","digital_certificate"],"languages":["en","es","ca"]}'
```
Comment (elboletaire) [inline .env.example:2]: This example does not match with the actual implementation.

File: vite/features.ts [hunk 5]
```diff
@@ -7,12 +7,26 @@ const features = () => {
...
+
   const features = merge.withOptions({ mergeArrays: false }, defaults, JSON.parse(process.env.FEATURES || '{}'))
+  const unimplemented_census = ['phone', 'email', 'crm', 'database', 'digital_certificate']
+  const unimplemented_voting_type = ['multi', 'approval', 'participatory', 'borda']
+
+  features.unimplemented_census.forEach((el) => {
+    if (!unimplemented_census.includes(el)) throw new Error('Unimplemented census ' + el + ' does not exist')
+  })
+  features.unimplemented_voting_type.forEach((el) => {
+    if (!unimplemented_voting_type.includes(el)) throw new Error('Unimplemented voting type ' + el + ' does not exist')
```
Comment (elboletaire) [inline vite/features.ts:28]: Try to use template strings for these errors.

**Review by elboletaire — CHANGES_REQUESTED** [review]

File: .env.example (hunk 4 above)
Comment (elboletaire) [reply .env.example]: @gerouvi I still miss the `unimplemented_voting_types`, `unimplemented_census` and `voting_types` in that FEATURES env 🤔

File: src/components/ProcessCreate/Steps/Census.tsx [hunk 6]
```diff
@@ -1,25 +1,35 @@
...
   const { t } = useTranslation()
+  const { isOpen, onOpen, onClose } = useDisclosure()
+
   const { form, setForm } = useProcessCreationSteps()
   const { defined, details } = useCensusTypes()
+  const { defined: definedUnim, details: detailsUnim } = useUnimplementedCensusTypes()
   const { censusType } = form
 
+  const [reason, setReason] = useState('')
+  const [showProCards, setShowProCards] = useState(false)
```
Comment (elboletaire) [inline src/components/ProcessCreate/Steps/Census.tsx:27]: No need to change it but as a FYI: for this kind of usage you can use the chakra-ui `useDisclosure` hook (used a few lines above actually). Simply rename the vars used to others and that's it.

**Review by elboletaire — APPROVED** [review]

**Pull request merged**